src/utils/pbconvert.hpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef PBCONVERT_HPP_
+#define PBCONVERT_HPP_
+
+#include <aos/common/types.hpp>
+
+#include <proto/servicemanager/v3/servicemanager.pb.h>
+
+/**
+ * Converts Aos string to PB string.
+ *
+ * @tparam cSize max PB string size.
+ * @param src Aos string to convert from.
+ */
+template <size_t cSize>
+constexpr void StringToPB(const aos::String& src, char (&dst)[cSize])
+{
+    aos::String(dst, cSize - 1) = src;
+}
+
+/**
+ * Converts PB string to Aos string.
+ *
+ * @tparam cSize max PB string size.
+ * @param dst Aos string to convert to.
+ */
+template <size_t cSize>
+constexpr void PBToString(const char (&src)[cSize], aos::String& dst)
+{
+    auto len = strlen(src);
+
+    if (len > cSize - 1) {
+        len = cSize - 1;
+    }
```

Comment:
I think it would be better to use assert here.

Reply:
Two points:
* if assert occurs - system is not usable any more;
* assert is enabled in debug mode only. For production, all assert will be disabled.

So I will add assert additionally to reducing size.

Reply:
ok